src/main/java/pl/gov/coi/pomocua/ads/BaseOffer.java

```diff
@@ -5,18 +5,16 @@
 import lombok.EqualsAndHashCode;
 import org.hibernate.validator.constraints.Length;
 
-import javax.persistence.Embedded;
-import javax.persistence.GeneratedValue;
-import javax.persistence.Id;
-import javax.persistence.MappedSuperclass;
+import javax.persistence.*;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.Pattern;
 
 import static javax.persistence.GenerationType.IDENTITY;
 
+@Entity
 @EqualsAndHashCode
 @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.PUBLIC_ONLY)
-@MappedSuperclass
+@Inheritance(strategy = InheritanceType.JOINED)
```

Comment:
When we discussed it, we said that `TABLE_PER_CLASS` would be suitable for us. Using `JOINED`, we return only `BaseOffer` fields in the `my-offers` endpoint without any subclass-specific fields, and I think that's not enough for the front-end (@sfc-gh-krzymkowski what do you think?)

Reply:
With `JOINED`, if we want to return all fields to the front-end, we have to... join other tables.
`TABLE_PER_CLASS` does union-query under the hood, and we can query for all types using `JpaRepository<BaseOffer>` (I think)